### DISCUSSION TITLE
M Series MAC Empty Debugger

### DISCUSSION THREAD

**Discussion by InitRoot** [Vector35/binaryninja-api#4864] [Q&A]
I'm having some issues with the debugger on my m series mac. The debug server settings and debug adapter settings are empty with nothing to select. I have no idea how to fix and if I need to install some dependencies etc.

![image](https://github.com/Vector35/binaryninja-api/assets/954507/b58aefbd-d164-4e5e-89f5-bcebdc8733d5)
![image](https://github.com/Vector35/binaryninja-api/assets/954507/e74be303-5323-4148-b840-46f138cb151a)

**Comment by xusheng6** [comment]
What is the binary you wish to debug?

**Reply by InitRoot** [reply]
For this one I had a x86 binary. Is this depended on the binary loaded?

**Reply by xusheng6** [reply]
Yes. What OS is the binary designed to run on?

**Reply by InitRoot** [reply]
Aaaw sorry, windows binary.

**Reply by xusheng6** [reply]
Well, no need to be sorry. 

This explains why the list is empty. However, this is also a bug, we should NOT show an empty list, since people have been hoping to debug a windows binary running inside of wine (via gdb).

**Reply by xusheng6** [reply]
See this issue: https://github.com/Vector35/debugger/issues

**Reply by InitRoot** [reply]
Thanks, i was also hoping windbg remote. Is there a way to get gdb to show then?

**Reply by xusheng6** [reply]
There is currently no way to do it. However, since this is relatively easy, I will consider prioritizing it. 

Windbg remote is in our mind: https://github.com/Vector35/debugger/issues/70. But given the complexity of it, it is unlikely to be dealt with soon. So you either debug it on a Windows machine, or, if I manage to make wine+gdb work, try debugging it within wine.

**Reply by InitRoot** [reply]
Excellent okay, then I'll wait on wine and gdb.
